packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -664,9 +664,10 @@ export class ConnectionManager implements IConnectionManager {
 	/**
 	 * Disconnect the current connection.
 	 * @param reason - Text description of disconnect reason to emit with disconnect event
+	 * @param error - Error causing the disconnect if any.
 	 * @returns A boolean that indicates if there was an existing connection (or pending connection) to disconnect
 	 */
-	private disconnectFromDeltaStream(reason: string): boolean {
+	private disconnectFromDeltaStream(reason: string, error?: IAnyDriverError): boolean {
```

Comment:
Looks like you still need to pass the error in here at the callsites where it's applicable (e.g. dispose and reconnect)

Reply:
I am passing the error below in reconnectOnError()

Reply:
Dispose is not that interesting imo as we already log the container close etc errors and we are not reconnecting in those cases, Here, we are reconnecting and the error causing the error could be interesting.